rustworkx-core/src/centrality.rs

```diff
@@ -335,6 +336,70 @@ fn accumulate_edges<G>(
         }
     }
 }
+/// Compute the degree centrality of all nodes in a graph.
+///
+/// The algorithm used in this function is based on:
+///
+/// Degree Centrality, Betweenness Centrality, and Closeness Centrality in Social Network
+/// Atlantis Press - Junlong Zhang, Yu Luo
+/// https://doi.org/10.2991/msam-17.2017.68
+///
+/// Example usage of `degree_centrality` function
+/// ```
+/// use rustworkx_core::petgraph::graph::UnGraph;
+/// use rustworkx_core::centrality::degree_centrality;
+///
+/// // Create an undirected graph with 4 nodes and 4 edges
+/// let graph = UnGraph::<i32, ()>::from_edges(&[
+///     (0, 1), (1, 2), (2, 3), (3, 0)
+/// ]);
+///
+/// // Calculate degree centrality
+/// let centrality = degree_centrality(&graph);
+///
+/// // Each node has 2 connections out of 3 possible (n-1 where n=4)
+/// for value in centrality {
+///     assert!((value - 2.0/3.0).abs() < 1e-10);
+/// }
+/// ```
+
+pub fn graph_degree_centrality<G>(graph: G) -> Vec<f64>
```

Comment:
For rustworkx-core, use a generic method that takes both kind of graphs. https://github.com/Qiskit/rustworkx/blob/aea56cf23b8a45716c1ae5af15a63b14222c8cf9/rustworkx-core/src/centrality.rs#L632 is a good example